specification/deviceprovisioningservices/Deviceprovisioningservices.Management/models.tsp

```diff
@@ -0,0 +1,872 @@
+ */
+model IotDpsPropertiesDescription {
+  /**
+   * Current state of the provisioning service.
+   */
+  state?: State;
+
+  /**
+   * Whether requests from Public Network are allowed
+   */
+  publicNetworkAccess?: PublicNetworkAccess;
+
+  /**
+   * The IP filter rules.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/missing-x-ms-identifiers" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  ipFilterRules?: IpFilterRule[];
+
+  /**
+   * Private endpoint connections created on this IotHub
+   */
+  privateEndpointConnections?: PrivateEndpointConnection[];
+
+  /**
+   * The ARM provisioning state of the provisioning service.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  provisioningState?: string;
+
+  /**
+   * List of IoT hubs associated with this provisioning service.
+   */
+  iotHubs?: IotHubDefinitionDescription[];
+
+  /**
+   * Allocation policy to be used by this provisioning service.
+   */
+  allocationPolicy?: AllocationPolicy;
+
+  /**
+   * Service endpoint for provisioning service.
+   */
+  @visibility(Lifecycle.Read)
+  serviceOperationsHostName?: string;
+
+  /**
+   * Device endpoint for this provisioning service.
+   */
+  @visibility(Lifecycle.Read)
+  deviceProvisioningHostName?: string;
+
+  /**
+   * Unique identifier of this provisioning service.
+   */
+  @visibility(Lifecycle.Read)
+  idScope?: string;
+
+  /**
+   * List of authorization keys for a provisioning service.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/missing-x-ms-identifiers" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  authorizationPolicies?: SharedAccessSignatureAuthorizationRuleAccessRightsDescription[];
+
+  /**
+   * Optional.
+   * Indicates if the DPS instance has Data Residency enabled, removing the cross geo-pair disaster recovery.
+   */
+  enableDataResidency?: boolean;
+
+  /**
+   * Portal endpoint to enable CORS for this provisioning service.
+   */
+  portalOperationsHostName?: string;
+}
+
+/**
+ * The IP filter rules for a provisioning Service.
+ */
+model IpFilterRule {
+  /**
+   * The name of the IP filter rule.
+   */
+  filterName: string;
+
+  /**
+   * The desired action for requests captured by this rule.
+   */
+  action: IpFilterActionType;
+
+  /**
+   * A string that contains the IP address range in CIDR notation for the rule.
+   */
+  ipMask: string;
+
+  /**
+   * Target for requests captured by this rule.
+   */
+  target?: IpFilterTargetType;
+}
+
+/**
+ * The properties of a private endpoint connection
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model PrivateEndpointConnectionProperties {
```

Comment:
>PrivateEndpointConnectionProperties [](http://example.com/codeflow?start=6&length=35)

feeling surprised this is not a part of PrivateEndpointConnection.tsp

Reply:
Yeah that is because the original definition is different from the on in common type.